packages/webapp/src/pages/Integrations/providerConfigKey/Settings/Show.tsx

```diff
@@ -41,6 +41,32 @@ export const SettingsSwitch: React.FC<{ data: GetIntegration['Success']['data'];
     }
 };
 
+const missingFieldsMessage = (data: GetIntegration['Success']['data']): string | null => {
+    switch (data.template.auth_mode) {
+        case 'OAUTH1':
+        case 'OAUTH2':
+        case 'TBA':
+            return settingsMissingOAuth(data.integration.missing_fields);
+
+        case 'APP':
+            return settingsMissingApp(data.integration.missing_fields);
+
+        case 'CUSTOM':
+            return settingsMissingCustom(data.integration.missing_fields);
+
+        case 'BASIC':
+        case 'API_KEY':
+        case 'APP_STORE':
+        case 'TABLEAU':
+        case 'NONE':
+        case 'OAUTH2_CC':
+            return null;
+
+        default:
+            return null;
+    }
+};
+
```

Comment:
can this function and the `settingsMissingX` ones be replaced by a map like this:
```
const FIELD_DISPLAY_NAMES = {
  OAUTH1: {
    oauth_client_id: 'Client ID',
    oauth_client_secret: 'Client Secret'
  },
  OAUTH2: {
    oauth_client_id: 'Client ID',
    oauth_client_secret: 'Client Secret'
  },
  TBA: {
    oauth_client_id: 'Client ID',
    oauth_client_secret: 'Client Secret'
  },
  APP: {
    oauth_client_id: 'App ID',
    oauth_client_secret: 'App Private Key',
    app_link: 'App Public Link'
  },
  CUSTOM: {
    oauth_client_id: 'Client ID',
    oauth_client_secret: 'Client Secret',
    app_link: 'App Public Link',
    app_id: 'App ID',
    private_key: 'App Private Key'
  }
} as const;

function missingFieldsMessage(data: GetIntegration['Success']['data']): string | null {
  const mappings = FIELD_DISPLAY_NAMES[data.template.auth_mode];
  if (!mappings) return null;
  
  return data.integration.missing_fields
    .map(field => mappings[field] || field)
    .join(', ');
};
```

Reply:
See c2f84977